gluten-flink/planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/common/source/VeloxSourceBuilder.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.flink.table.planner.plan.nodes.exec.common.source;
+
+import org.apache.gluten.streaming.api.operators.GlutenStreamSource;
+import org.apache.gluten.table.runtime.operators.GlutenValuesSourceFunction;
+import org.apache.gluten.util.LogicalTypeConverter;
+import org.apache.gluten.util.PlanNodeIdGenerator;
+import org.apache.gluten.util.ReflectUtils;
+
+import io.github.zhztheplayer.velox4j.connector.VectorConnectorSplit;
+import io.github.zhztheplayer.velox4j.connector.VectorTableHandle;
+import io.github.zhztheplayer.velox4j.plan.StatefulPlanNode;
+import io.github.zhztheplayer.velox4j.plan.TableScanNode;
+
+import org.apache.flink.api.dag.Transformation;
+import org.apache.flink.streaming.api.transformations.LegacySourceTransformation;
+import org.apache.flink.table.connector.source.ScanTableSource;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.runtime.typeutils.InternalTypeInfo;
+import org.apache.flink.table.types.logical.RowType;
+import org.apache.flink.types.Row;
+import org.apache.flink.util.FlinkRuntimeException;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+public class VeloxSourceBuilder {
+
+  public static Transformation<RowData> build(
+      Transformation<RowData> transformation, ScanTableSource scanTableSource) {
+    if (transformation instanceof LegacySourceTransformation) {
+      if (scanTableSource.getClass().getSimpleName().equals("TestValuesScanLookupTableSource")) {
+        return buildVectorSource(transformation, scanTableSource);
+      }
+    }
+    return transformation;
+  }
+
+  @SuppressWarnings({"rawtypes", "unchecked"})
+  private static Transformation<RowData> buildVectorSource(
+      Transformation<RowData> transformation, ScanTableSource tableSource) {
+    LegacySourceTransformation<RowData> sourceTransformation =
+        (LegacySourceTransformation<RowData>) transformation;
+    try {
+      Class<?> tableSourceClazz =
+          Class.forName(
+              "org.apache.flink.table.planner.factories.TestValuesTableFactory$TestValuesScanTableSourceWithoutProjectionPushDown");
```

Comment:
This name is under test since in test environment it loads this class? Which class it use in production environment?

Reply:
This is specially designed for test，and do not have a class in production.